napari/_qt/updates.py

```diff
@@ -0,0 +1,350 @@
+import sys
+import uuid
+from pathlib import Path
+from time import sleep
+
+from qtpy.QtCore import QDir, QFile, QObject, QProcess, Signal
+
+from ..utils.notifications import notification_manager
+from ..utils.translations import trans
+from .qthreading import create_worker
+
+
+class ManagerActions:
+    install = "install"
+    update = "update"
+    clear = "clear"
+    clean = "clean"
+    remove = "remove"
+
+
+class UpdateManager(QObject):
+    """Manager for handling the update process.
+
+    Parameters
+    ----------
+    parent : QObject, optional
+        Parent of the manager. Default is None.
+    """
+
+    started = Signal()
+    finished = Signal()
+    errored = Signal(str)
+
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self._finished = True
+        self._process = None
+        self._processes = []
+        self._worker_thread = None
+        self._messages = []
+        # FIXME: add keys per action
+        self._update_keys = [
+            ('conda-forge/noarch', trans._('update - downloading')),
+            ('Updating specs:', trans._('update - downloading')),
+            ('Total download:', trans._('update - downloading')),
+            ('Preparing transaction:', trans._('update - installing')),
+            ('Verifying transaction:', trans._('update - installing')),
+            ('Executing transaction:', trans._('update - installing')),
+            ('To activate this environment', trans._('update - installing')),
+        ]
+
+    def _handle_yield(self, total):
+        """Handle yielded progress for the dummy process."""
+        i, msg = self._current_progress[-1]
+        pbar = self._worker_thread.pbar
+        pbar.set_description(msg)
+        # This is a hack to get the progress bar to update when runnning a
+        # QProcess
+        pbar.total = (
```

Comment:
I'm not sure if this is working? When I run `Clean` from the troubleshoot dialog I first get a short indeterminate progress bar and then a bar that doesn't update. The number of total iterations changes but not the number of completed iterations.

![image](https://user-images.githubusercontent.com/17995243/166081350-620dc997-db2a-4d03-888a-b018d25165de.png)

I also think you might want to be using `pbar.update`? I tried that but it just gives me an indeterminate bar so I'm not sure I'm following the logic.

Reply:
Hi @DragaDoncila I removed the dialog for this version we only have an entry on the `File` menu, unfortunately the clean command does not have much output so the progress bar will not get much feedback :-\